legend-engine-config/legend-engine-repl/legend-engine-repl-client/src/main/java/org/finos/legend/engine/repl/client/Client.java

```diff
@@ -289,8 +296,7 @@ private void persistHistory()
 
     public Path getHomeDir()
     {
-        String homeDir = System.getProperty("legend.repl.storageDir");
-        return FileUtils.getUserDirectory().toPath().resolve(homeDir != null ? homeDir : ".legend/repl");
+        return this.homeDirectory;
```

Comment:
I think there's benefit in allowing configuring the home dir via a CLI config flag. Many users will boot the REPL from a JAR - they don't write a custom `main()` and invoke a `Client` with custom path, so I think let's keep the previous behavior, and the System property/CLI flag should have higher precedent than the parameter being passed in the `Client` constructor

```suggestion
        return System.getProperty("legend.repl.homeDir") != null ? System.getProperty("legend.repl.homeDir") : this.homeDirectory;
```